shapely/strtree.py

```diff
@@ -34,317 +34,185 @@ class STRtree:
         """
-        result = self._tree.query(geom, 0)
-        return self.geometries[result]
-
-    def query(self, geom: BaseGeometry) -> Sequence[BaseGeometry]:
-        """Query for nodes which intersect the geom's envelope to get
-        geometries corresponding to the items stored in the nodes.
-
-        This method is an alias for query_geoms. It may be removed in
-        version 2.0.
-
-        Parameters
-        ----------
-        geom : geometry object
-            The query geometry.
-
-        Returns
-        -------
-        An array of geometry objects.
-
-        """
-        return self.query_geoms(geom)
-
-    def query_bulk(self, geometry, predicate=None, distance=None):
-        """Returns all combinations of each input geometry and geometries in the tree
-        where the envelope of each input geometry intersects with the envelope of a
-        tree geometry.
-
-        If predicate is provided, a prepared version of each input geometry
-        is tested using the predicate function against each item whose
-        extent intersects the envelope of the input geometry:
-        predicate(geometry, tree_geometry).
+        Return the integer indices of all combinations of each input geometry
+        and tree geometries where the extent of each input geometry intersects
+        the extent of a tree geometry.
+
+        If input geometry is a scalar, this returns an array of shape (n, ) with
+        the indices of the matching tree geometries.  If input geometry is an
+        array_like, this returns an array with shape (2,n) where the subarrays
+        correspond to the indices of the input geometries and indices of the
+        tree geometries associated with each.  To generate an array of pairs of
+        input geometry index and tree geometry index, simply transpose the
+        results.
+
+        If a predicate is provided, the tree geometries are further filtered to
+        those that meet the predicate when comparing the input geometry to the
+        tree geometry:
+        predicate(geom, tree_geometry)
 
         The 'dwithin' predicate requires GEOS >= 3.10.
 
-        This returns an array with shape (2,n) where the subarrays correspond
-        to the indexes of the input geometries and indexes of the tree geometries
-        associated with each.  To generate an array of pairs of input geometry
-        index and tree geometry index, simply transpose the results.
-
-        In the context of a spatial join, input geometries are the "left" geometries
-        that determine the order of the results, and tree geometries are "right" geometries
-        that are joined against the left geometries.  This effectively performs
-        an inner join, where only those combinations of geometries that can be joined
-        based on envelope overlap or optional predicate are returned.
+        In the context of a spatial join, input geometries are the "left"
```

Comment:
I agree that it would be good to split this out, but I'm unclear where the notes section would go if not in the docstring?

I'm thinking eventually we may want a handwritten page in the docs for STRtree rather than just the docstrings, where we can give more verbose examples around custom items and geometries; would this notes section go there?  (if so, I'd rather cut content from here and deal with that in a follow-on PR)

Reply:
Ah, the "Notes" section is still part of the docstring (it's one of the recognized section titles by numpydoc), it's just a section that is typically put below the "Parameters" section (or also below the "Examples" section).

(and indeed, eventually it would be good to have a user guide page about the strtree)